core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -1118,6 +1126,24 @@ public boolean isForNewerHudson() {
             }
         }
 
+        /**
+         * Returns true iff the plugin declares a minimum Java version and it's newer than what the Jenkins master is running on.
+         *
+         * @return true iff the plugin declares a minimum Java version and it's newer than what the Jenkins master is running on.
+         *
+         * @since TODO
+         */
+        public boolean isForNewerJava() {
+            try {
+                return minimumJavaVersion != null && new VersionNumber(minimumJavaVersion).isNewerThan(
+                        new VersionNumber(System.getProperty("java.specification.version")));
+            } catch (NumberFormatException nfe) {
+                logBadMinJavaVersion("'minimumJavaVersion' was specified for plugin ''{0}'' but unparseable (received {1})", this.name,
```

Comment:
Can you set the message inside the method avoiding passing it as an argument?. It's always the same

Reply:
Yeah, I considered it, but didn't do it because this seemed not like a big deal. Looking.